examples/reader-test.kl

```diff
@@ -0,0 +1,63 @@
+#lang "prelude.kl"
```

Comment:
Should I put those examples in `reader.kl` instead? I created it because I saw a bunch of similarly-named `*-test.kl` files, but I now realize that those are all exceptions for tests which can't be written in the same file as the code they exercise.